enarx.ks

```diff
@@ -90,10 +90,88 @@ Type=ether
 [Network]
 DHCP=yes
 EOF
+
+get_maj_min() {
+    local _majmin
+    _majmin="$(stat -L -c '%t:%T' "$1" 2>/dev/null)"
+    printf "%s" "$((0x${_majmin%:*})):$((0x${_majmin#*:}))"
+}
+
+# give the /srv luks partition a partition label
+if ! [[ -b /dev/disk/by-partlabel/srv ]]; then
+   luksdev=/dev/$(ls -1 /sys/dev/block/$(get_maj_min /dev/disk/by-label/srv)/slaves)
+
+   # label the srv partition for later
+   sfdisk --part-label \
+       /dev/block/$(cat /sys/dev/block/$(get_maj_min $luksdev)/../dev) \
+       $(cat /sys/dev/block/$(get_maj_min "$luksdev")/partition) \
+       srv
+else
+   luksdev=/dev/disk/by-partlabel/srv
+fi
+
+# add /srv to /etc/fstab on reinstall
+if ! grep -q "/srv" /etc/fstab; then
+   echo "LABEL=srv /srv xfs defaults 0 0" >> /etc/fstab
+
+   # reconstruct /etc/crypttab
+   for i in /dev/disk/by-uuid/*; do
+       [[ -b "$i" ]] || continue;
+       [[ "$i" -ef "$luksdev" ]] || continue
+       uuid=${i#/dev/disk/by-uuid/}
+       echo "luks-$uuid UUID=$uuid none discard" > /etc/crypttab
+       break
+   done
+
+else
+
+   # umount /srv and close the luks device
+   if umount /srv &>/dev/null || ! mountpoint /srv &>/dev/null; then
+      cryptsetup close /dev/disk/by-label/srv &>/dev/null || :
+   fi
+
+   # bind /srv to the tpm
+   if ! clevis luks unlock -d "$luksdev"; then
+      clevis luks bind -f -k- -d "$luksdev" tpm2 '{}' <<< "temppass"
```

Comment:
Are we sure this uses the right PCR?

Reply:
I don't know.. do we need to specify a PCR?

Reply:
We probably should be explicit here. Otherwise we might end up with unrecoverable data.